dvc/command/config.py

```diff
@@ -136,6 +135,27 @@ def _check(self, conf, remote, section, opt=None):
                 f"option '{opt}' doesn't exist in {name} '{section}'"
             )
 
+    def _get_appropriate_levels(self, levels):
+        if levels:
+            self._validate_level_for_non_repo_operation(levels)
+            return [self.args.level]
```

Comment:
This should either be using `self.args.level` everywhere, or the `level` function parameter everywhere, it should not mix the two when doing the validation + return

```suggestion
    def _get_appropriate_levels(self, level):
        if level:
            self._validate_level_for_non_repo_operation(level)
            return [level]
```